### ISSUE TITLE
SC2096 incorrectly warns about shebang when "env -S" is used

### ISSUE THREAD

**Issue by devjgm** [koalaman/shellcheck#2105]
#### For bugs
- SC2096 
- online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2096)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

Related to: https://github.com/koalaman/shellcheck/issues/833

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/env -S bash -x
# shellcheck shell=bash
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
Line 1:
#!/usr/bin/env -S bash -x
^-- SC2096: On most OS, shebangs can only specify a single parameter.
```

#### Here's what I wanted or expected to see:

Shellcheck should be aware of `env -S`, which according to https://man7.org/linux/man-pages/man1/env.1.html

```
       -S, --split-string=S
              process and split S into separate arguments; used to pass
              multiple arguments on shebang lines
```

According to https://ss64.com/osx/env.html, `env -S` also should work on macOS.

**Comment by koalaman** [comment]
Definitely true. This has now been fixed, and shell detection should work properly for such shebangs. Thanks!

**Comment by kartynnik** [comment]
`env -S` should be suggested as a solution on the Wiki for SC2096. Is it better to create a new issue for that?
